### PR TITLE
Added methods for Top and Hot subreddit sections

### PR DESCRIPTION
**Pull request by MajorMJR**
I've added some methods to get the top posts from a subreddit and the hot posts.

### CODE REVIEW

**Comment by jzelinskie** [comment]
Thanks for the PR! I'll probably have some time to look this over in a next few days.
One thing was apparent -- can you change the file permissions back to 644?

**Comment by MajorMJR** [comment]
Hey sorry about that, I've corrected the file permissions now.

File: submission.go [hunk 1]
```diff
@@ -53,3 +55,182 @@ func (h *Submission) String() string {
...
+		submissions[i] = child.Data
+	}
+
+	return submissions, nil
+}
+
+// Returns top submmissions from a subreddit
+// Second argument must have value of one of "hour", "day", "week", "month", "year", "all"
+func (s Session) TopSubmissions(subreddit string, time string) ([]*Submission, error) {
+	times := []string{"hour", "day", "week", "month", "year", "all"}
```
Comment (jzelinskie) [inline submission.go]: Maybe these should be their own enum type so people are left guessing what strings to use

**Comment by jzelinskie** [comment]
I think we should keep these definitions in the session.go file. I have kept all methods on the session type in there just for sanity's sake.

**Comment by MajorMJR** [comment]
Yeah I agree, I moved them there as I thought the session file was getting too large but it makes more sense to keep them in the session file.

I'm not sure what you mean with creating an enum type.  Something like this: http://stackoverflow.com/questions/14426366/what-is-an-idiomatic-way-of-representing-enums-in-golang ?

I should have some time to make these changes later this evening.

**Comment by jzelinskie** [comment]
Yep something like:

```go
// Probably needs a better name
type time string

const (
	Hour  time = "hour"
	Day   time = "day"
	Week  time = "week"
	Month time = "month"
	Year  time = "year"
	All   time = "all"
)
```

This way the type documents the possible values, rather than simply "string".

It's debatable, because if reddit added a new value, the library would specifically have to be patched to use it. However, this makes it easier for someone to simply open up this package and not have to know how to use the reddit API before doing so.

**Comment by MajorMJR** [comment]
Thanks, yeah I like that solution much cleaner that way.

**Comment by MajorMJR** [comment]
Hey sorry for the delay, been busy with work this week.  

I've been attempting to add in the enum type but I'm not sure if I'm going about this the right way.  If you get a chance could you check out my latest commit and see where I could improve?  thanks

File: session.go [hunk 2]
```diff
@@ -89,41 +175,24 @@ func (s Session) SubredditSubmissions(subreddit string) ([]*Submission, error) {
...
-			return nil, fmt.Errorf("cannot sort %s by %s", popularity, age)
+// Returns top submmissions from a subreddit
+// Second argument must have value of one of "hour", "day", "week", "month", "year", "all"
+func (s Session) TopSubmissions(subreddit string, t PostAge) ([]*Submission, error) {
+	times := []string{"hour", "day", "week", "month", "year", "all"}
+	url := ""
+	timeValid := ""
+
+	//check if time is valid value
+	for _, k := range times {
```
Comment (jzelinskie) [inline session.go]: You don't need this loop. If you do not export the `PostAge` type (i.e. `type postAge string`), but export the possible values, (i.e. `const Hour postAge = "hour"`), there are no possible ways of creating more of the type `reddit.postAge` besides the ones exported by this library, thus it becomes impossible for a user of this method to pass an invalid `postAge`.

**Comment by jzelinskie** [comment]
@MajorMJR ping

**Comment by MajorMJR** [comment]
Hey Jzelinskie,

I finally got around to finishing up this code, but it looks like the latest commit by aggrolite solves the problem much cleaner than my solution.  

Sorry about the long delay, I'll keep looking for something I can improve.

**Pull request closed without merging**